apps/xprof_gui/priv/src/actions/MonitoringActions.js

```diff
@@ -38,23 +38,16 @@ export const stopMonitoringFunction = monitored => async (
   }
 };
 
-export const startMonitoringFunction = (
-  functionName,
-  onSuccess,
-  onError,
-) => async (dispatch, getState) => {
-  const state = getState();
-  const monitoredCollection = getAllMonitored(state);
-  const isMonitored = monitoredCollection.filter(f => f.query === functionName)
-    .length;
-
-  let error;
-  if (!isMonitored) {
-    ({ error } = await XProf.startMonitoringFunction(functionName));
-  }
-  if (error && onError) {
-    onError(error);
-  } else if (onSuccess) {
-    onSuccess();
+export const startMonitoringFunction = query => async (dispatch) => {
```

Comment:
I would keep previous API of this function (onSucces parameter, but without OnError - it will be common - adding notification). We also execute this function on `calleeClick`, and in such case we don't want to add 'clicked' function to recent queries and clear function browser.

Reply:
actually my intention was to add the clicked function also to history. This way history is all functions that were monitored sometimes. (You can scroll back to a clicked function and edit/expand/finetune the query.)

But maybe there is some other consideration, to keep the original API and add a utility funcition with the common onSuccess and onError handlers?

Reply:
If it is intentional we don't need to keep current API :)